### DISCUSSION TITLE
Compile under Rosetta on M1 machine

### DISCUSSION THREAD

**Discussion by louy2** [Homebrew/discussions#2540] [Everyday usage] [answered]
I am installing a formula from a tap, from source, which gives me error saying this architecture is not supported, which is kind of expected on M1. Now I have already upgraded my homebrew, is there a way to run the compilation in Rosetta?

**Comment by carlocab** [comment, marked as answer]
You can either run the install script for Homebrew under Rosetta to install it in `/usr/local` and use that installation of Homebrew, or edit the formula to use `arch -x86_64` at the start of each `system` call in the `install` method. The latter may not work, depending on the formula.

**Reply by louy2** [reply]
I am given this error:

    Error: Cannot install in Homebrew on ARM processor in Intel default prefix (/usr/local)!

**Reply by carlocab** [reply]
Is that from running `/usr/local/bin/brew`? You'll need to run that under Rosetta as well.

You may wish to create an alias that runs
```
arch -x86_64 /usr/local/bin/brew
```

**Reply by louy2** [reply]
I succeeded using `/usr/local/bin/brew` after entering a `arch -x86_64 zsh` subshell.
